src/Arcus.EventGrid/Contracts/CloudOrEventGridEvent.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Linq;
+using Arcus.EventGrid.Contracts.Interfaces;
+using CloudNative.CloudEvents;
+using GuardNet;
+using Microsoft.Azure.EventGrid.Models;
+
+namespace Arcus.EventGrid.Contracts
+{
+    /// <summary>
+    /// Representation of a supported event in Azure Event Grid.
+    /// This can either be an <see cref="CloudEvent"/> or <see cref="EventGridEvent"/> instance.
+    /// </summary>
+    /// <remarks>
+    ///     This model is not build for custom events.
+    ///     Create your own event by inheriting from <see cref="EventGridEvent{TData}"/>.
+    /// </remarks>
+    public sealed class CloudOrEventGridEvent : IEvent
```

Comment:
Maybe rename to `Event` or so? In this case we don't care so let's just leave it out of the name maybe?

Reply:
We have already such a type with that name.

Reply:
How does that one relate to this one again? For custom events probably? 🤔 

I'm just not a fan of this one so thinking what a better name could be.

Reply:
Yeah, the one we have with the name `Event` is now obsolete; and refers to using `EventGridEvent`  instead.

Reply:
Given we're working on a new major version I would then already remove Event and use this one instead. What do you think?

Reply:
Let me check where this obsolete type is being used first.

Reply:
`SubscriptionValidation` is the only implementation that uses it (also obsolete). But since all the other obsolete events are already removed; I see no problem with removing this one too.

Let me change that for you to rename `CloudOrEventGridEvent` > `Event`.

Reply:
Where-ever it's used, kill it with fire!

Reply:
Done.

Reply:
😂 Thanks! ❤️